clash-lib/src/Clash/Driver.hs

```diff
@@ -720,36 +725,42 @@ compilePrimitive idirs pkgDbs topDir
         Just f -> pure f
         Nothing -> do
           r <- loadImportAndInterpret idirs iArgs topDir qualMod funcName "TemplateFunction"
-          processHintError (show bbGenName) pNm id r
+          processHintErrors (show bbGenName) pNm r
     pure (BBFunction (Data.Text.unpack pNm) hsh tf)
 
 compilePrimitive _ _ _ (Primitive pNm wf typ) =
   return (Primitive pNm wf typ)
 {-# SCC compilePrimitive #-}
 
-processHintError
-  :: Monad m
-  => String
-  -> Data.Text.Text
-  -> (t -> r)
```

Comment:
Note for reviewers: there's a slight refactor in here. I've removed the `(t -> r)` argument. Is now handled by the caller.